src/solvers/dg_curved/dg_3d.jl

```diff
@@ -147,6 +155,161 @@ end
 end
 
 
+# TODO: Taal dimension agnostic
+function calc_boundary_flux!(cache, u, t, boundary_condition::BoundaryConditionPeriodic,
+                             equations, mesh::CurvedMesh{3}, dg::DG)
+  @assert isperiodic(mesh)
+end
+
+
+function calc_boundary_flux!(cache, u, t, boundary_condition,
+                             equations, mesh::CurvedMesh{3}, dg::DG)
```

Comment:
I've probably asked this before, but why doesn't this method just call
```julia
calc_boundary_flux!(cache, u, t, (boundary_condition,), equations, mesh::CurvedMesh{3}, dg::DG)
```
i.e., wraps the single function in a tuple such that the other method is selected and be done with it? That way you could save a lot of redundancy (also for 1D/2D). But maybe I'm missing something...

Reply:
I implemented this in 2D, but you preferred having a lot of duplicate code (#520).

Reply:
What I meant was the following: You have one method `calc_boundary_flux!` with signature
```julia
function calc_boundary_flux!(cache, u, t, boundary_condition, equations, mesh::CurvedMesh{3}, dg::DG) 
...
end
```
and one with
```julia
function calc_boundary_flux!(cache, u, t, boundary_conditions::Union{NamedTuple,Tuple}, equations, mesh::CurvedMesh{3}, dg::DG)
...
end
```

Now, the second version is the more general one (allowing different BCs in each direction), thus you could implement the first one as follows:
```julia
function calc_boundary_flux!(cache, u, t, boundary_condition, equations, mesh::CurvedMesh{3}, dg::DG) 
   calc_boundary_flux!(cache, u, t,
                       (boundary_condition, boundary_condition, boundary_condition,
                        boundary_condition, boundary_condition, boundary_condition),
                       equations, mesh, dg)
end
```
There is still one implementation for each dimension (see the dispatch on `CurvedMesh{3}`), but you eliminate one redundant function in each dimension. Do you know what I mean? And does it make sense?